src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -458,10 +458,21 @@ internal static string GetSharedListenerIdOrNull(ITriggerBinding triggerBinding)
 
         internal static void CheckRetrySupport(ITriggerBinding triggerBinding, FunctionDescriptor descr, ILogger logger)
         {
-            if (descr.RetryStrategy != null && triggerBinding != null && triggerBinding.GetType().GetCustomAttribute<SupportsRetryAttribute>() == null)
+            if (descr.RetryStrategy != null)
             {
-                logger?.LogWarning($"Retries are not supported by the trigger binding for { descr.ShortName}.");
-                descr.RetryStrategy = null; // Need to fix tests
+                string functionName = descr?.ShortName;
+                string triggerBindingName = triggerBinding?.GetType().Name;
+                string strategy = JsonConvert.SerializeObject(descr?.RetryStrategy);
+
+                logger?.LogDebug("Retry strategy is defined for function '{FunctionName}' with trigger binding type 'TriggerBindingName'. Strategy = '{Strategy}'", 
```

Comment:
Thanks for going structured here! Needs just a touch of formatting love for the middle arg:
```suggestion
                logger?.LogDebug("Retry strategy is defined for function '{FunctionName}' with trigger binding type '{TriggerBindingName}'. Strategy = '{Strategy}'", 
```

Reply:
If the goal of this new log is to just measure retry usage, it seems the LogFunctionRetryAttempt logging we already do on the execution path is sufficient for that? I.e. let's just query that existing log rather than add a new one.

Reply:
we also want to log:
1. Retry is specified but there were not retry executions.
2. Number of retries specified by trigger type.

Reply:
My issue with putting this log in the SDK indexing like this is that it establishes a pattern we don't want at this layer. Note that there are lots of features like Singleton, Timeout, etc. that users can use for functions, and we don't log the status of each one on startup.

This telemetry is only needed for Azure Functions, and the established pattern for this is to log metric events (logged to FunctionsMetrics table). E.g. you can see an example [here](https://github.com/Azure/azure-functions-host/blob/ebe1aafa9bf01ed56202b21a1a47a1897d2cebf9/src/WebJobs.Script.WebHost/Diagnostics/FunctionInstanceLogger.cs#L78) where we log binding type usage metrics. We should do something similar for retry metrics.

Reply:
Tagging @divyagandhisethi as FYI as the logging approach agreed here needs to be leveraged in building detectors